### PR TITLE
Prevent ingesting samples older than `past_grace_period`

### PR DESCRIPTION
**Pull request by colega**
#### What this PR does

This PR implements a new limit, similar to `creation_grace_period`, that limits the age of the ingested samples based on the wall clock.

It also takes into account the OOO window, to avoid having to change both settings when OOO window is modified.

The default value of 1 week should cover any observability needs. If some customer wants to backfill their samples in order, they can increase the value of that to `100y` in the config to cover 100 years, for example.

#### Which issue(s) this PR fixes or relates to

Fixes #<issue number>

#### Checklist

- [x] Tests updated.
- [x] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

File: pkg/ingester/lifecycle_test.go
Comment (pstibrany) [inline pkg/ingester/lifecycle_test.go:33]: We can just disable it, right?

File: pkg/util/validation/limits.go
Comment (pstibrany) [inline pkg/util/validation/limits.go:258]: super-nit: `ooo` -> `OOO`?

File: pkg/ingester/lifecycle_test.go
Comment (colega) [reply pkg/ingester/lifecycle_test.go:33]: I thought about using `0` value as `disabled`, but something inside of me didn't like it. I think it's fair to expect some kind of config set up.

Comment (pstibrany) [reply pkg/ingester/lifecycle_test.go:33]: Yeah, that's what I was suggesting. We can set reasonable default (once it's tested), but having ability to disable would be nice.

Comment (colega) [reply pkg/ingester/lifecycle_test.go:33]: So you're suggesting to use zero value to disable it?

File: pkg/util/validation/limits.go
Comment (colega) [reply pkg/util/validation/limits.go]: Did this in some commit below. Thanks.

File: pkg/ingester/lifecycle_test.go
Comment (colega) [reply pkg/ingester/lifecycle_test.go]: After talking on Slack, I changed the behavior to interpret 0 as disabled.

**Comment by colega** [comment]
I'll wait for @jdbaldry's review on the runbooks before merging.

File: docs/sources/mimir/manage/mimir-runbooks/_index.md
Comment (jdbaldry) [inline docs/sources/mimir/manage/mimir-runbooks/_index.md:1724]: We prefer "refer to" over "see"

```suggestion
Refer to [`err-mimir-too-far-in-past`](#err-mimir-too-far-in-past) for more details and how to fix it.
```

**Review by jdbaldry — APPROVED** [review]
Docs LGTM and are consistent with the rest of the runbook format. 

nit: consider using contractions to match our tone guidelines: https://grafana.com/docs/writers-toolkit/write/style-guide/voice-tone-guidelines/#tone

File: docs/sources/mimir/manage/mimir-runbooks/_index.md
Comment (colega) [inline docs/sources/mimir/manage/mimir-runbooks/_index.md:1709]:
```suggestion
- The samples that are too far in the past aren't ingested.
```

**Pull request merged**
